### ISSUE TITLE
Enterprise deprecated

### ISSUE THREAD

**Issue by beffge** [goadesign/model#502]
The [enterprise](https://github.com/goadesign/model/blob/4c1dc02adce64280fbe732d31dc16ddcd4cbb4ae/mdl/model.go#L12) is deprecated in the DSL and should be replaced with [groups](https://github.com/structurizr/dsl/blob/master/docs/language-reference.md#enterprise).

Could someone hint to all the places which might need changes for it to work?

**Comment by raphael** [comment]
Thanks for raising this! A good ol' search and replace should work pretty well here given that Enterprise is used mostly as metadata. Roughly the DSL has to be added (similar to [this](https://github.com/goadesign/model/blob/main/dsl/design.go#L107-L127)), then the group expression needs to be added (that the new `group` DSL would instantiate, similar to [this](https://github.com/goadesign/model/blob/main/mdl/model.go#L12)), the [ModelizeDesign](https://github.com/goadesign/model/blob/main/mdl/eval.go#L32) function in the `mdl` package needs to be updated to take this new field into consideration. We should also probably test to make sure that the design [serializes](https://github.com/goadesign/model/blob/main/stz/client.go#L92) correctly for the structurizr service.

**Comment by raphael** [comment]
@beffge nudge: any chance you might be able to take a look?

**Comment by beffge** [comment]
Hey, sorry for responding so slow on your comment. I can't promise anything here timeline wise but it is till on my radar.

**Comment by beffge** [comment]
I just thought I'll look into this again. But I got a bit confused.

To me it looks more like the DeploymentEnv i.e. in the resulting JSON just a string and during modelling as a function. Difference is just, it can be at different locations as well as it can be nested.

With regards to being nested, it requires
```
        properties {
            "structurizr.groupSeparator" "/"
        }
```
for which I didn't find anything similar. How would you approach this?

**Comment by beffge** [comment]
OK, I tried to add it but got lost 😓 It seems like, since group can be part of `Design`, `SoftwareSystem` and `Container` as well as nested, it's more way more than just the existing `Enterprise`. I started adding it, but I was just assuming what I had to do, since I don't use anything but the `mdl` package. I'm pretty sure any PR I'd create would require a lot of discussion and rework i.e. it will be faster and more efficient if someone which uses the "other" parts implements it.

**Comment by raphael** [comment]
Fair enough! I'll keep the issue opened so one of us can take a look at some point.
